blockreassurance.php

```diff
@@ -1,404 +1,512 @@
+        $link = 'https://addons.prestashop.com' . $parsedUrl['path'] . '?' . http_build_query($parameters);
+        $categoryFetcher = array(
+            'name' => 'Blocks, Tabs and Banners',
+            'link' => $link,
+            'description' => 'With these modules, you can personalize your e-commerce website by adding reassurance blocks (free delivery, satisfaction guaranteed or your money back, etc.), illustrating your categories with visuals on your homepage or adding advertising banners to your PrestaShop store.',
+        );
 
-            return $html.$helper->generateForm($this->fields_form);
-        } elseif (Tools::isSubmit('deleteblockreassurance')) {
-            $reassurance = new reassuranceClass((int)$id_reassurance);
-            if (file_exists(dirname(__FILE__).'/img/'.$reassurance->file_name)) {
-                unlink(dirname(__FILE__).'/img/'.$reassurance->file_name);
-            }
-            $reassurance->delete();
-            $this->_clearCache('*');
-            Tools::redirectAdmin(AdminController::$currentIndex.'&configure='.$this->name.'&token='.Tools::getAdminTokenLite('AdminModules'));
-        } else {
-            $content = $this->getListContent((int)Configuration::get('PS_LANG_DEFAULT'));
-            $helper = $this->initList();
-            $helper->listTotal = count($content);
-            return $html.$helper->generateList($content, $this->fields_list);
+        $id_lang = $this->context->language->id;
+
+        $currentPage = 'global';
+        $getPage = Tools::getValue('page');
+        if (!empty($getPage)) {
+            $currentPage = $getPage;
         }
 
-        if (isset($_POST['submitModule'])) {
-            Configuration::updateValue('BLOCKREASSURANCE_NBBLOCKS', ((isset($_POST['nbblocks']) && $_POST['nbblocks'] != '') ? (int)$_POST['nbblocks'] : ''));
-            if ($this->removeFromDB() && $this->addToDB()) {
-                $this->_clearCache('blockreassurance.tpl');
-                $output = '<div class="conf confirm">'.$this->trans('The block configuration has been updated.', array(), 'Modules.Blockreassurance.Admin').'</div>';
-            } else {
-                $output = '<div class="conf error"><img src="../img/admin/disabled.gif"/>'.$this->trans('An error occurred while attempting to save.', array(), 'Admin.Notifications.Error').'</div>';
-            }
+        $moduleAdminLink = Context::getContext()->link->getAdminLink('AdminModules', true) . '&configure=' . $this->name . '&module_name=' . $this->name;
+
+        $allCms = CMS::listCms($id_lang);
+
+        $this->context->smarty->assign(array(
+            'addons_category' => $categoryFetcher,
+            'psr_hook_header' => Configuration::get('PSR_HOOK_HEADER'),
+            'psr_hook_footer' => Configuration::get('PSR_HOOK_FOOTER'),
+            'psr_hook_product' => Configuration::get('PSR_HOOK_PRODUCT'),
+            'psr_hook_checkout' => Configuration::get('PSR_HOOK_CHECKOUT'),
```

Comment:
Having this configuration data is not necessary.

We could implement instead the widget system and display the same content everywhere.